src/main/java/net/dv8tion/jda/api/entities/MessageReference.java

```diff
@@ -0,0 +1,234 @@
+     *     <br>The request was attempted after the channel was deleted.</li>
+     * </ul>
+     *
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *          If this reference refers to a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the logged in account does not have
+     *         <ul>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ}</li>
+     *             <li>{@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}</li>
+     *         </ul>
+     *
+     * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.api.entities.Message}
+     */
+    @Nonnull
+    public RestAction<Message> resolve()
+    {
+        checkPermission(Permission.MESSAGE_READ);
+        checkPermission(Permission.MESSAGE_HISTORY);
+
+        Route.CompiledRoute route = Route.Messages.GET_MESSAGE.compile(getChannelId(), getMessageId());
+        JDAImpl jda = (JDAImpl) getJDA();
+
+        return new RestActionImpl<>(jda, route,
+                (response, request) -> jda.getEntityBuilder().createMessage(response.getObject(), getChannel(), false));
+    }
+
+    /**
+     * Referenced message.
+     *
+     * <p>This will have different meaning depending on the {@link Message#getType() type} of message.
+     * Usually, this is a {@link MessageType#INLINE_REPLY INLINE_REPLY} reference.
+     * This can be null even if the type is {@link MessageType#INLINE_REPLY INLINE_REPLY}, when the message it references doesn't exist or discord wasn't able to resolve it in time.
+     *
+     * @return The referenced message, or null
+     */
+    @Nullable
+    public Message getMessage()
+    {
+        return referencedMessage;
+    }
+
+    /**
+     * The channel from which this message originates.
+     * <br>Messages from other guilds can be referenced, in which case JDA may not have the channel cached.
+     *
+     * @return The origin channel for this message reference.
+     */
+    @Nonnull
+    public MessageChannel getChannel()
+    {
+        return channel;
+    }
+
+
+    /**
+     * The guild for this reference.
+     * <br>
+     * This will be null if the message did not come from a guild, or JDA did not have the guild cached
+     *
+     * @return The guild, or null.
+     */
+    @Nullable
+    public Guild getGuild()
+    {
+        return guild;
+    }
+
+    /**
+     * Returns the message id for this reference.
+     *
+     * @return The message id
+     */
+    public long getMessageIdLong()
+    {
+        return messageId;
+    }
+
+    /**
+     * Returns the channel id for this reference.
+     *
+     * @return The channel id
+     */
+    public long getChannelIdLong()
+    {
+        return channelId;
+    }
+
+    /**
+     * Returns the guild id for this reference.
+     * @return The guild id
+     */
+    public long getGuildIdLong()
```

Comment:
Is this always present?

Reply:
Not when sent in DMs. The docs also state that message_id is not present on [Channel Follow Add](https://discord.com/developers/docs/resources/channel#message-types-channel-follow-add-messages).

 How should JDA handle the getters in this case? Would nullable Long be an acceptable solution?

Reply:
I would return 0 and explicitly document this behavior